データベース設計

evetakumatsu opened the pull request:
# What
データベース設計の記述

# Why
まず、必要なテーブル、エンティティを洗い出し、ER図を作成した。
<img src="https://gyazo.com/57e2465d9e8d9c88128bdfc2e18e0f52.png" alt="画像の代替テキスト">
このER図を元に、READMEにデータベース設計を記述した。

evetakumatsu committed:
データベース設計

ER図で書いたデータベース設計を、READMEに記述

sh10n7 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| column             | type       | options                        |
+| ------------------ | ---------- | ------------------------------ |
+| nickname           | string     | null: false                    |
+| email              | string     | null: false, UNIQUE            |
```
```
| email | string | null: false, UNIQUE |
```
一意制制約を記述する場合は、```unique: true```と記載しましょう！

evetakumatsu replied:
コードを修正しました。

sh10n7 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| first_name         | string     | null: false                    |
+| kana_last_name     | string     | null: false                    |
+| kana_first_name    | string     | null: false                    |
+| date_of_birth      | data       | null: false                    |
```
```
| date_of_birth | data | null: false |
```
こちらは日付を保存するカラムですので、Typeは**日付型**を使用しましょう。

evetakumatsu replied:
コードを修正しました。

sh10n7 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| status             | string     | null: false                    |
+| delivery_charge    | string     | null: false                    |
+| ship_from          | string     | null: false                    |
+| days_to_ship       | string     | null: false                    |
```
```
| category           | string     | null: false                    |
| status             | string     | null: false                    |
| delivery_charge    | string     | null: false                    |
| ship_from          | string     | null: false                    |
| days_to_ship       | string     | null: false                    |
```
以下2点変更しましょう
・Typeは**integer**に変更しましょう。
・カラム名は**○○_id**という形に変更しましょう。

理由：
都道府県の選択にはActivehashを利用する実装予定となっております。
このとき保存される値は、id番号であり整数値であるためです。

evetakumatsu replied:
コードを修正しました。

sh10n7 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+### Association
 
-* How to run the test suite
+- has_one    :orders
```
```
- has_one :orders
```
こちらordersは単数に変更しましょう。

理由：
アソシエーションの記述は
has_oneの場合、所有先のモデルは単数形にしないと正しく動作しないからです。

evetakumatsu replied:
コードを修正しました。

sh10n7 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| column             | type       | options                        |
+| ------------------ | ---------- | ------------------------------ |
+| postal             | string     | null: false                    |
+| prefectures        | string     | null: false                    |
```
```
| prefectures | string | null: false |
```
配送先のカラムは、itemsテーブルの発送元の地域のカラムと同じものを使用しましょう！
itemsテーブルの配送元カラムをprefecture_idに修正いただいても構いません。

【理由】
今回はアクティブハッシュを使用し都道府県情報をmodelファイルに用意します。
そしてmodelファイルを呼び出せば、複数の箇所で都道府県情報が使用できるようになるためです。

evetakumatsu replied:
コードを修正しました。

sh10n7 commented on README.md:
```diff
@@ -1,24 +1,67 @@
+| municipalities     | string     | null: false                    |
+| address            | string     | null: false                    |
+| building           | string     |                                |
+| tel                | varchar    | null: false                    |
```
```
| tel | varchar | null: false |
```
電話番号カラムのTypeは**string**に変更しましょう。

理由：
0から始まる電話番号を登録するためです。

evetakumatsu replied:
コードを修正しました。

sh10n7 reviewed (commented):
レビュー依頼ありがとうございます。
修正をお願いしておりますので、ご確認いただけますと幸いです。

なお、修正後はコミット・プッシュを確認いただき、再度**修正後のER図**を添付の上、**レビュー依頼フォーム**のご提出をお願いいたします。

ご不明点や疑問点ございましたら、通話メンターにご相談ください！
どうぞ、よろしくお願いいたします。

evetakumatsu committed:
一意性制約の修正

'email'カラムの一意性制約の修正

evetakumatsu committed:
カラムの型変更

’date_of_birth’カラムの型を’datetime ’に変更

evetakumatsu committed:
itemsテーブルの設定変更

これらの変更したカラムはActivehashを利用するため、idに変更し、型もintegerに変更した。

evetakumatsu committed:
itemテーブルの関連性の設定変更

アソシエーションの記述を修正した。

evetakumatsu committed:
sending_addressテーブルの設定変更

’Activehash’を利用するため、カラムを’item’テーブルと同じ都道府県、’ship_from_id’に変更した。
'tel'カラムも型も修正した。

OhyaYouki reviewed (commented):
レビュー依頼ありがとうございます！

修正点を確認いたしました。

**LGTM**とさせていただきます！
実装お疲れ様でした！

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に
次の実装のためのブランチを切って次の課題に取り組みましょう！

引き続き、開発頑張っていきましょう！応援しております。
